### PR TITLE
feat: support testing on bun

### PR DESCRIPTION
**Pull request by achingbrain**
Run node tests with bun using env var `NODE_EXEC=bun`

### CODE REVIEW

**Comment by achingbrain** [comment]
Bah, it's broken on Windows, support might be added in a week or so - https://twitter.com/bunjavascript/status/1753267105909416079

**Review by SgtPooki — APPROVED** [review]
~good with this. Do we want to wait and see on bun+windows before merge?~

we should wait on this until bun+windows.. https://github.com/ipfs-examples/test-ipfs-example/actions/runs/7831891366/job/21369332451?pr=18

**Comment by salieflewis** [comment]
> ~good with this. Do we want to wait and see on bun+windows before merge?~
> 
> we should wait on this until bun+windows.. https://github.com/ipfs-examples/test-ipfs-example/actions/runs/7831891366/job/21369332451?pr=18

Bun supports Windows 10 and later as of 1.1

https://bun.sh/blog/bun-v1.1

File: test/bun.spec.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe('bun', function () {
+  // slow CI is slow
+  this.timeout(540000)
+  this.slow(60000)
+
+  if (platform() === 'win32') {
+    it.skip('bun does not run on windows', () => {})
+
+    return
+  }
```
Comment (SgtPooki) [inline test/bun.spec.ts]:
```suggestion
```

**Review by SgtPooki — CHANGES_REQUESTED** [review]

File: test/bun.spec.ts (hunk 1 above)
Comment (SgtPooki) [reply test/bun.spec.ts]: testing changes in https://github.com/ipfs-examples/test-ipfs-example/pull/19

**Pull request merged**